assignments/array-methods.js

```diff
@@ -2,82 +2,652 @@
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal = runners.map(function(element) {
+	return element.donation;
+});
+
+let ticketPriceTotalSum = ticketPriceTotal.reduce(function(
+	accumulator,
+	currentValue
+) {
+	return (accumulator += currentValue);
+},
+0);
+
+//ARROW FUNCTION
+// ticketPriceTotal = runners.map(element => element.donation);
+// ticketPriceTotalSum = ticketPriceTotal.reduce((accumulator, currentValue) => {
+// 	return (accumulator += currentValue);
+// }, 0);
+
 console.log(ticketPriceTotal);
+console.log(ticketPriceTotalSum);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//We want to thank all the runners for their donations and we have decided to give them a present based on their donations. If donated less than 10$, it is going to be pen. If donated less than 100$ it is going to be a hat, if donated more than 100$ it is going to be tshirt. Filter runners into the 3 lists based on their donations.
+let donationLessThan10 = [];
+let donationLessThan100 = [];
+let donationMoreThan100 = [];
+
+donationLessThan10 = runners.filter(function(element) {
+	return element.donation < 10;
+});
+donationLessThan100 = runners.filter(function(element) {
+	return element.donation < 100;
+});
+donationMoreThan100 = runners.filter(function(element) {
+	return element.donation >= 100;
+});
+
+//ARROW FUNCTION
+// donationLessThan10 = runners.filter(element => element.donation < 10);
+// donationLessThan100 = runners.filter(element => element.donation < 100);
+// donationMoreThan100 = runners.filter(element => element.donation >= 100);
+
+console.log(donationLessThan10);
+console.log(donationLessThan100);
+console.log(donationMoreThan100);
 
 // Problem 2
+//The Tshirts company told you to size up every tshirt as their tshirts are super small. You did that and you need to update the information in the list.
+let newTshirtsSizes = [];
+
+newTshirtsSizes = runners.map(function(element) {
+	let newShirtSize = "";
+	if (element.shirt_size === "XS") {
+		newShirtSize = "S";
+	} else if (element.shirt_size === "S") {
+		newShirtSize = "M";
+	} else if (element.shirt_size === "M") {
+		newShirtSize = "L";
+	} else if (element.shirt_size === "L") {
+		newShirtSize = "XL";
+	} else if (element.shirt_size === "XL") {
+		newShirtSize = "XXL";
+	} else {
+		newShirtSize = element.shirt_size;
+	}
+
+	//ARROW FUNCTION
+	// newTshirtsSizes = runners.map(element => {
+	// 	let newShirtSize = "";
+	// 	if (element.shirt_size === "XS") {
+	// 		newShirtSize = "S";
+	// 	} else if (element.shirt_size === "S") {
```

Comment:
rather than just commenting out stretch problems. create another branch name it stretch . The whole idea of git is to create multiple copies of the same thing.